integrations/docker/images/chip-cert-bins/Dockerfile

```diff
@@ -0,0 +1,198 @@
+            && sh cmake-3.19.3-Linux-aarch64.sh --exclude-subdir --prefix=/usr/local \
+            && rm -rf cmake-3.19.3-Linux-aarch64.sh) \
+            && exec bash \
+            ;; \
+        *) ;; \
+    esac
+
+# Python 3.9 and PIP
+RUN set -x \
+    && DEBIAN_FRONTEND=noninteractive  apt-get update \
+    && DEBIAN_FRONTEND=noninteractive  apt-get install -y libgirepository1.0-dev \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y software-properties-common \
+    && add-apt-repository universe \
+    && curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py \
+    && python3.9 get-pip.py \
+    && update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 1 \
+    && rm -rf /var/lib/apt/lists/ \
+    && : # last line
+
+RUN set -x \
+    && pip3 install attrs coloredlogs PyGithub pygit future portpicker mobly click cxxfilt ghapi pandas tabulate \
+    && : # last line
+
+# build and install gn
+RUN set -x \
+    && git clone https://gn.googlesource.com/gn \
+    && cd gn \
+    && python3 build/gen.py \
+    && ninja -C out \
+    && cp out/gn /usr/local/bin \
+    && cd .. \
+    && rm -rf gn \
+    && : # last line
+
+# Install bloat comparison tools
+RUN set -x \
+    && git clone https://github.com/google/bloaty.git \
+    && mkdir -p bloaty/build \
+    && cd bloaty/build \
+    && cmake ../ \
+    && make -j8 \
+    && make install \
+    && cd ../.. \
+    && rm -rf bloaty \
+    && : # last line
+
+# NodeJS: install a newer version than what apt-get would read
+# This installs the latest LTS version of nodejs
+RUN case ${TARGETPLATFORM} in \
+        "linux/amd64") \
+            set -x \
+			&& mkdir node_js \
+			&& cd node_js \
+			&& wget https://nodejs.org/dist/v12.19.0/node-v12.19.0-linux-x64.tar.xz \
+			&& tar xfvJ node-v12.19.0-linux-x64.tar.xz \
+			&& mv node-v12.19.0-linux-x64 /opt/ \
+			&& ln -s /opt/node-v12.19.0-linux-x64 /opt/node \
+			&& ln -s /opt/node/bin/* /usr/bin \
+			&& cd .. \
+			&& rm -rf node_js \
+			;; \
+        "linux/arm64")\
+            set -x \
+			&& mkdir node_js \
+			&& cd node_js \
+			&& wget https://nodejs.org/dist/v12.19.0/node-v12.19.0-linux-arm64.tar.xz \
+			&& tar xfvJ node-v12.19.0-linux-arm64.tar.xz \
+			&& mv node-v12.19.0-linux-arm64 /opt/ \
+			&& ln -s /opt/node-v12.19.0-linux-arm64 /opt/node \
+			&& ln -s /opt/node/bin/* /usr/bin \
+			&& cd .. \
+			&& rm -rf node_js \
+			;; \
+        *) ;; \
+    esac
+
+# Set python to python3
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 10
+
+RUN mkdir /root/connectedhomeip
+ADD ./ /root/connectedhomeip/
+WORKDIR /root/connectedhomeip/
+RUN scripts/build/gn_bootstrap.sh
+RUN gn gen out/debug --args='chip_mdns="platform" chip_inet_config_enable_ipv4=false'
+RUN ninja -C out/debug
+
+RUN scripts/examples/gn_build_example.sh examples/all-clusters-app/linux/ examples/all-clusters-app/linux/out/all-clusters-app chip_inet_config_enable_ipv4=false
```

Comment:
why are you using gn_build_example.sh and not build_examples.py? that one has a much more standardized output format and can generate archive output as well.

Reply:
build_examples.py does not have support for all of the targets, such as the tv-casting-app and app1 simulated device. The ladder of which I could not get to build using the build_examples.py. I could go back and try to add in the missing targets if you feel strongly about this.